### PR TITLE
wasmserve: init at 1.0.0

### PR DESCRIPTION
**Pull request by kirillrdy**
###### Description of changes

https://github.com/hajimehoshi/wasmserve
is just simple wasm server for golang wasm apps, somewhat similar to trunk from rust world.

I used it now and then and maybe its useful to others.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/wasmserve/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{ buildGoModule
+, lib
+, fetchFromGitHub
+}:
+buildGoModule {
+  pname = "wasmserve";
+  version = "2022-11-29";
```
Comment (ocfox) [inline pkgs/servers/wasmserve/default.nix:7]:
```suggestion
  version = "unstable-2022-11-29";
```

File: pkgs/servers/wasmserve/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    rev = "689d7ac772c6be55a8deb81885898db78ba81651";
+    sha256 = "sha256-t8QLquAedj57DzE1/8NthxM+r4Hd2GdYTP/sO0XrxHw=";
+  };
+  vendorHash = null;
+  meta = with lib; {
+    description = "An HTTP server for testing Wasm";
+    homepage = "https://github.com/hajimehoshi/wasmserve";
+    license = licenses.asl20;
+    maintainers = [ maintainers.kirillrdy ];
+    platforms = platforms.all;
```
Comment (ocfox) [inline pkgs/servers/wasmserve/default.nix:20]: This should have been set by buildGoModule by default.

**Review by ocfox — CHANGES_REQUESTED** [review]

File: pkgs/servers/wasmserve/default.nix (hunk 1 above)
Comment (kirillrdy) [reply pkgs/servers/wasmserve/default.nix]: done

File: pkgs/servers/wasmserve/default.nix (hunk 2 above)
Comment (kirillrdy) [reply pkgs/servers/wasmserve/default.nix]: removed

**Comment by kirillrdy** [comment]
@ocfox  I am also thinking of moving this from pkgs/server to pkgs/development/tools same place where trunk lives

**Review by ocfox — APPROVED** [review]

**Comment by Janik-Haag** [comment]
Result of `nixpkgs-review pr 228659` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>wasmserve</li>
  </ul>
</details>

File: pkgs/development/tools/wasmserve/default.nix [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    sha256 = "sha256-KlCbUre6yIorE1ZM++Rto8vgwVGsC1wZj1xCd3AwQy0=";
+  };
+  vendorHash = null;
+  meta = with lib; {
+    description = "An HTTP server for testing Wasm";
+    homepage = "https://github.com/hajimehoshi/wasmserve";
+    license = licenses.asl20;
+    maintainers = [ maintainers.kirillrdy ];
+  };
+}
```
Comment (Janik-Haag) [inline pkgs/development/tools/wasmserve/default.nix:21]:
```suggestion
{ lib
, buildGoModule
, fetchFromGitHub
}:

buildGoModule rec {
  pname = "wasmserve";
  version = "1.0.0";

  src = fetchFromGitHub {
    owner = "hajimehoshi";
    repo = "wasmserve";
    rev = "v${version}";
    hash = "sha256-KlCbUre6yIorE1ZM++Rto8vgwVGsC1wZj1xCd3AwQy0=";
  };

  vendorHash = null;

  meta = with lib; {
    description = "An HTTP server for testing Wasm";
    homepage = "https://github.com/hajimehoshi/wasmserve";
    license = licenses.asl20;
    maintainers = with maintainers; [ kirillrdy ];
  };
}
Commenting
```

just some formatting nitpicks. 
btw do you know [nix-init](https://github.com/nix-community/nix-init)? would have probably work great for this.

**Review by Janik-Haag — CHANGES_REQUESTED** [review]
just some formatting nitpicks mostly looks good seems to work fine.

File: pkgs/development/tools/wasmserve/default.nix (hunk 3 above)
Comment (kirillrdy) [reply pkgs/development/tools/wasmserve/default.nix]: formatting changes make sense to me,

I have seen nix-init although never used it, will give it a try ! thanks !

File: pkgs/development/tools/wasmserve/default.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+buildGoModule rec {
+  pname = "wasmserve";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "hajimehoshi";
+    repo = "wasmserve";
+    rev = "v${version}";
+    sha256 = "sha256-KlCbUre6yIorE1ZM++Rto8vgwVGsC1wZj1xCd3AwQy0=";
```
Comment (Janik-Haag) [inline pkgs/development/tools/wasmserve/default.nix:14]:
```suggestion
    hash = "sha256-KlCbUre6yIorE1ZM++Rto8vgwVGsC1wZj1xCd3AwQy0=";
```

File: pkgs/development/tools/wasmserve/default.nix [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  };
+
+  vendorHash = null;
+
+  meta = with lib; {
+    description = "An HTTP server for testing Wasm";
+    homepage = "https://github.com/hajimehoshi/wasmserve";
+    license = licenses.asl20;
+    maintainers = [ maintainers.kirillrdy ];
```
Comment (Janik-Haag) [inline pkgs/development/tools/wasmserve/default.nix:23]:
```suggestion
    maintainers = with maintainers; [ kirillrdy ];
```

**Review by Janik-Haag — COMMENTED** [review]
should be good after that.

**Comment by Janik-Haag** [comment]
Result of `nixpkgs-review pr 228659` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>wasmserve</li>
  </ul>
</details>

**Review by Janik-Haag — APPROVED** [review]

**Pull request merged**